src/test/mir-opt/simplify-locals-fixedpoint/rustc.foo.SimplifyLocals.diff

```diff
@@ -0,0 +1,67 @@
+          StorageDead(_3);                 // bb0[7]: scope 0 at $DIR/simplify-locals-fixedpoint.rs:4:68: 4:69
+          StorageDead(_2);                 // bb0[8]: scope 0 at $DIR/simplify-locals-fixedpoint.rs:4:68: 4:69
+          _5 = discriminant((_1.0: std::option::Option<u8>)); // bb0[9]: scope 0 at $DIR/simplify-locals-fixedpoint.rs:4:13: 4:20
+          switchInt(move _5) -> [1isize: bb1, otherwise: bb3]; // bb0[10]: scope 0 at $DIR/simplify-locals-fixedpoint.rs:4:13: 4:20
+      }
+  
+      bb1: {
+          _4 = discriminant((_1.1: std::option::Option<T>)); // bb1[0]: scope 0 at $DIR/simplify-locals-fixedpoint.rs:4:22: 4:26
+          switchInt(move _4) -> [0isize: bb2, otherwise: bb3]; // bb1[1]: scope 0 at $DIR/simplify-locals-fixedpoint.rs:4:22: 4:26
+      }
+  
+      bb2: {
+          StorageLive(_6);                 // bb2[0]: scope 0 at $DIR/simplify-locals-fixedpoint.rs:4:18: 4:19
+          _6 = (((_1.0: std::option::Option<u8>) as Some).0: u8); // bb2[1]: scope 0 at $DIR/simplify-locals-fixedpoint.rs:4:18: 4:19
+-         StorageLive(_7);                 // bb2[2]: scope 1 at $DIR/simplify-locals-fixedpoint.rs:5:12: 5:20
+-         StorageLive(_8);                 // bb2[3]: scope 1 at $DIR/simplify-locals-fixedpoint.rs:5:12: 5:13
+-         _8 = _6;                         // bb2[4]: scope 1 at $DIR/simplify-locals-fixedpoint.rs:5:12: 5:13
+-         _7 = Gt(move _8, const 42u8);    // bb2[5]: scope 1 at $DIR/simplify-locals-fixedpoint.rs:5:12: 5:20
+-                                          // ty::Const
+-                                          // + ty: u8
+-                                          // + val: Value(Scalar(0x2a))
+-                                          // mir::Constant
+-                                          // + span: $DIR/simplify-locals-fixedpoint.rs:5:16: 5:20
+-                                          // + literal: Const { ty: u8, val: Value(Scalar(0x2a)) }
+-         StorageDead(_8);                 // bb2[6]: scope 1 at $DIR/simplify-locals-fixedpoint.rs:5:19: 5:20
+-         StorageDead(_7);                 // bb2[7]: scope 1 at $DIR/simplify-locals-fixedpoint.rs:8:5: 8:6
+-         StorageDead(_6);                 // bb2[8]: scope 0 at $DIR/simplify-locals-fixedpoint.rs:8:5: 8:6
+-         goto -> bb3;                     // bb2[9]: scope 0 at $DIR/simplify-locals-fixedpoint.rs:4:5: 8:6
++         StorageDead(_6);                 // bb2[2]: scope 0 at $DIR/simplify-locals-fixedpoint.rs:8:5: 8:6
++         goto -> bb3;                     // bb2[3]: scope 0 at $DIR/simplify-locals-fixedpoint.rs:4:5: 8:6
+      }
+  
+      bb3: {
+          drop(_1) -> bb4;                 // bb3[0]: scope 0 at $DIR/simplify-locals-fixedpoint.rs:9:1: 9:2
+      }
+  
+      bb4: {
+          StorageDead(_1);                 // bb4[0]: scope 0 at $DIR/simplify-locals-fixedpoint.rs:9:1: 9:2
+          return;                          // bb4[1]: scope 0 at $DIR/simplify-locals-fixedpoint.rs:9:2: 9:2
```

Comment:
Hmm, just staring at this: could we normalize `$DIR/test-name.rs` to `$FILE`?
It would cause a big `src/test/ui` diff overall but it might be worth it.